python/tests/experimental/api/test_logger.py

```diff
@@ -154,11 +154,13 @@ def test_log_batch_ranking_metrics_default_target():
     assert pandas_summary.loc["top_rank", "counts/n"] == 1
     assert pandas_summary.loc["average_precision_k_" + str(k), "counts/n"] == 1
     assert pandas_summary.loc["norm_dis_cumul_gain_k_" + str(k), "counts/n"] == 1
+    # ndcg@3 = [0.9013]
+    assert isclose(pandas_summary.loc[f"norm_dis_cumul_gain_k_{k}", "distribution/median"], 0.90130, abs_tol=0.00001)
 
 
 def test_log_batch_ranking_metrics_ranking_ndcg_wikipedia():
     # From https://en.wikipedia.org/wiki/Discounted_cumulative_gain#Example
-    ranking_df = pd.DataFrame({"targets": [[3, 2, 3, 0, 1, 2, 3, 2]], "predictions": [[7, 6, 5, 4, 3, 2, 1, 0]]})
+    ranking_df = pd.DataFrame({"targets": [[3, 2, 3, 0, 1, 2, 3, 2]], "predictions": [[3, 2, 3, 0, 1, 2]]})
```

Comment:
I find the terminology confusing here, these aren't really targets and predictions, it seems what we need is a single column of relevancies for the ndcg calculation when they are numeric. Do we use both the predictions and targets here?

Reply:
yes, it is confusing - Let's say we have a list of documents D_0 to D_7. The `predictions` col tells us that the recommendation system recommended [D_0,D_1,D_2,D_3,D_4,D5], and their relevance (obtained, e.g., by asking humans) is [3,2,3,0,1,2]. But we have two extra documents in our pool that were not suggested - D_6 and D_7. Those two are in the `targets` col, which contains the relevance for all documents.

So, both predictions and target are used here - in particular, the order in the predicitons col is important to calculate DCG, and targets is important because it contains the relevances for all documents, even those not recommended, which is used to calculate IDCG.

@bernease please check if that matches your understanding on how `predictions` and `targets` should be passed.